gym/vector/vector_env.py

```diff
@@ -211,6 +217,42 @@ def seed(self, seed=None):
             "Please use `env.reset(seed=seed) instead in VectorEnvs."
         )
 
+    def _add_info(self, infos: dict, info: dict, env_num: int) -> dict:
+        """Add env info to the info dictionary of the vectorized environment.
+
+        Given the `info` of a single environment add it to the `infos` dictionary
+        which represents all the infos of the vectorized environment.
+        Every `key` of `info` is paired with a boolean mask `_key` representing
+        whether or not the i-indexed environment has this `info`.
+
+        Args:
+            infos (dict): the infos of the vectorized environment
+            info (dict): the info coming from the single environment
+            env_num (int): the index of the single environment
+
+        Returns:
+            infos (dict): the (updated) infos of the vectorized environment
+
+        """
+        for k in info.keys():
+            if k not in infos:
+                info_array, array_mask = self._init_info_arrays(type(info[k]))
+            else:
+                info_array, array_mask = infos[k], infos[f"_{k}"]
+
+            info_array[env_num], array_mask[env_num] = info[k], True
+            infos[k], infos[f"_{k}"] = info_array, array_mask
+        return infos
+
+    def _init_info_arrays(self, dtype: type) -> np.ndarray:
+        if dtype in [int, float, bool] or issubclass(dtype, np.number):
+            array = np.zeros(self.num_envs, dtype=dtype)
+        else:
+            array = np.zeros(self.num_envs, dtype=object)
```

Comment:
Was there a discussion somewhere already whether it should be initialized as `0` as opposed to `nan`? Intuitively I prefer `nan` because a `0` can have an actual meaning. Especially since for `object` we're using None. And in the end you should use the mask to index it away anyways.

Reply:
Here: https://github.com/openai/gym/pull/2773#discussion_r859582951
Basically we decided for 0 since the `_key` can be used for masking whether is a 0 coming from not having data or a 0 with an actual meaning. There was also some discussion in the issue regarding the initialization: https://github.com/openai/gym/issues/2657#issuecomment-1055597901

Reply:
Some more context also here: https://github.com/google/brax/issues/194